server/routes/notification.js

```diff
@@ -0,0 +1,90 @@
+const authMiddleware = require("../middlewares/authMiddleware");
+const ListProductModel = require("../models/listProductModel");
+const NotificationModel = require("../models/notificationModel");
+const ProductModel = require("../models/productModel");
+const UserListModel = require("../models/userListModel");
+const UserNotificationModel = require("../models/userNotificationsModel");
+const router = require("express").Router();
+
+//Make the given notification id read
+router.get("/read/:id", authMiddleware, async (req, res) => {
+  try {
+    const { id } = req.params;
+    const { userId } = req.headers;
+
+    if (!id) return res.status(400).send({ msg: "Must have an id parameter" });
+    if (!userId)
+      return res.status(400).send({ msg: "Must have a userId in headers" });
+
+    const userNotification = await UserNotificationModel.findOne({
+      where: { notification_id: id, user_id: userId },
+    });
+
+    const updatedUserNotification = await userNotification.update({
+      isRead: true,
+    });
+
+    res.send(updatedUserNotification);
+  } catch (error) {
+    res.status(500).send({
+      msg: "Server Error",
+    });
+  }
+});
+
+//Create a notification using a product id given inside the body
+router.post("/createNotification", async (req, res) => {
```

Comment:
A notification will only happen on a price drop.

Reply:
I understand. To clarify, I was wondering why there would specifically be a route for this, since I'm not sure this is functionality the user should be interfacing with.

Reply:
My knowledge might be lacking on this one lol. What would be a good way to make this inaccessible by a user and only by a scraping service?

Reply:
Not sure on how are services are going to be implemented. That's probably going to be Sina's decision. If I were to guess we would probably set up scraping to happen at certain times of the day, but it's not something I think we would need a router for.

Reply:
If I'm understanding this correctly, we wont be using a route for creating the model since it will be accessible by a user?

Reply:
That would be my guess. Although the code you wrote in the route would still most likely get used.

Reply:
This is a good point. Notifications can be handled via service as well. This is okay for now, it can easily be refactored into a service if we need to by just copy pasting the code lol